conans/client/rest/rest_client.py

```diff
@@ -241,12 +242,17 @@ def upload_package(self, package_reference, the_files, retry, retry_wait):
 
     @handle_return_deserializer()
     def authenticate(self, user, password):
-        '''Sends user + password to get a token'''
+        """Sends user + password to get a token"""
         auth = HTTPBasicAuth(user, password)
         url = "%s/users/authenticate" % self._remote_api_url
         t1 = time.time()
         ret = self.requester.get(url, auth=auth, headers=self.custom_headers,
                                  verify=self.verify_ssl)
+        if ret.status_code == 401:
+            raise AuthenticationException("Wrong user or password")
+        # Cannot check content-type=text/html, conan server is doing it wrong
+        if not ret.ok or "html>" in ret.content:
+            raise ConanException("Invalid server response, check remote URL and try again")
```

Comment:
Not display HTML? Maybe it is useful (as if it is a proxy). If not, maybe say something int he error message about proxies too.

Why "html>" and not "<html>"?

Reply:
We could display the html, and the error at the end? if the html is printed, then the screen gets full of crappy code. 
About the tag, "html>" covers also end and begin, but begin could have attributes like <html foo="bar"